thin-replica-server/proto/state_snapshot_concord.proto

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 VMware, all rights reserved
+//
+// Concord Client's StateSnapshot Service
+
+syntax = "proto3";
+package com.vmware.concord.state_snapshot_concord;
+
+option java_package = "com.vmware.concord.state_snapshot_concord";
+
+// The Concord StateSnapshot service can be used by the concord client to 
+// retrieve state snapshot from a concord cluster
+// See `client/proto/state_snapshot.proto` for definition of state snapshot.
+
+service StateSnapshot {
+  // Stream a specific state snapshot in a resumable fashion as a finite stream of key-values.
+  // Key-values are streamed with lexicographic order on keys.
+  // Errors:
+  // NOT_FOUND: if a state snapshot with the requested ID is not (or no longer) available.
+  //            Applications are advised to retry initialization by fetching a new snapshot ID.
+  // INVALID_ARGUMENT: if resuming a stream using `from_key` and that key is not part of the
+  //                   state snapshot.
+  // UNAVAILABLE: if Concord Client is not ready yet to process requests.
+  // FAILED_PRECONDITION: if a precondition in Concord Client is not satisfied. For example,
+  //                      the state might be corrupted or invalid and, in that case, there
+  //                      would be no point in proceeding with streaming.
+  // UNKNOWN: exact cause is unknown.
+  rpc StreamSnapshot(StreamSnapshotRequest) returns (stream StreamSnapshotResponse);
```

Comment:
If its a bidirectional stream, then the request should also have a stream keyword. A stream represents a sequence of message exchange in GRPC, it's generally a good practice not to name the Request or Response as a stream. Can we use Replay instead of Stream.
So if its a server-side streaming then we might say:
rpc ReplaySnapshot(SnapshotRequest) returns (stream SnapshotResponse);

And if its a bidirectional stream we can use:
rpc ReplaySnapshot(stream SnapshotRequest) returns (stream SnapshotResponse);

Reply:
The API is most likely modeled after `state_snapshot.proto`.
It is not a bi-directional stream, instead, you send one request and as a response you get multiple (finite amount of) answers.

If we want to get rid of the word `Stream` then I suggest the following:
`rpc GetSnapshot(GetSnapshotRequest) returns (stream GetSnapshotResponse)` or
`rpc GetSnapshot(GetSnapshotRequest) returns (stream KeyValuePairs)`

If we change the naming here then we should also change it in `state_snapshot.proto`. @dartdart26

Reply:
I prefer we don't change it, because this interface is used in many places, affecting different teams.